### PR TITLE
Cv 596 display price message

### PR DESCRIPTION
**Pull request by hkf-tech**

### CODE REVIEW

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Views/Apprentice/ConfirmChangeOfEmployer.cshtml [hunk 1]
```diff
@@ -3,13 +3,25 @@
...
 }
 
 <div class="govuk-grid-row">
     <div class="govuk-grid-column-two-thirds">
         <h1 class="govuk-heading-xl">@ViewBag.Title</h1>
+
+        @if (Model.ExceedsFundingBandCap)
+        {
+            <div class="govuk-inset-text das-inset-text--highlight">
+                <h3 class="govuk-heading-s">The apprenticeship is above the funding band maximum</h3>
```
Comment (chrisfoster76) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Views/Apprentice/ConfirmChangeOfEmployer.cshtml]: Should say `This apprenticeship`...

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Views/Apprentice/ConfirmChangeOfEmployer.cshtml [hunk 2]
```diff
@@ -3,13 +3,25 @@
...
+                <p>
+                    The new agreed apprenticeship price is above the @DisplayCost(Model.FundingBandCap.Value) <a target="_blank" href="https://www.gov.uk/government/publications/apprenticeship-funding-bands">maximum value for the funding band</a> for this apprenticeship.
```
Comment (chrisfoster76) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Views/Apprentice/ConfirmChangeOfEmployer.cshtml]: Should say `value of the funding band`

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Views/Apprentice/ConfirmChangeOfEmployer.cshtml [hunk 3]
```diff
@@ -3,13 +3,25 @@
...
+                    You'll need to ask the new employer to pay the difference directly to you - this can be funded from their account.
```
Comment (chrisfoster76) [inline src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Views/Apprentice/ConfirmChangeOfEmployer.cshtml]: Should say `can't be funded from their account`

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/ApprenticeController.cs [hunk 4]
```diff
@@ -148,7 +156,7 @@ public async Task<IActionResult> Price(PriceRequest request)
         public async Task<IActionResult> Price(PriceViewModel viewModel)
         {
             var request = await _modelMapper.Map<ChangeOfEmployerRequest>(viewModel);
-            return RedirectToRoute(RouteNames.ApprenticeConfirmChangeOfEmployer, request);
+            return RedirectToRoute(RouteNames.ApprenticeConfirmChangeOfEmployer, new { request.ProviderId, request.ApprenticeshipHashedId, request.EmployerAccountLegalEntityPublicHashedId, request.StartDate, request.Price });
```
[the remark this answers is not in the document]
Comment (hkf-tech) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/ApprenticeController.cs]: the action changed to confirm and likewise the request, view model, mapper classes and test.

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Apprentice/ChangeOfEmployerViewModelMapper.cs [hunk 5]
```diff
@@ -45,5 +54,45 @@ public async Task<ChangeOfEmployerViewModel> Map(ChangeOfEmployerRequest source)
...
+
+        private async Task<(GetApprenticeshipResponse Apprenticeship,
+           GetPriceEpisodesResponse PriceEpisodes,
+           AccountLegalEntityResponse AccountLegalEntity,
+           ITrainingProgramme TrainingProgramme)>
+           GetApprenticeshipData(long apprenticeshipId)
+        {
+            var apprenticeship = await _commitmentApiClient.GetApprenticeship(apprenticeshipId);
+            var priceEpisodesTask = _commitmentApiClient.GetPriceEpisodes(apprenticeshipId);
+            var legalEntityTask =  _commitmentApiClient.GetLegalEntity(apprenticeship.AccountLegalEntityId);
```
[the remark this answers is not in the document]
Comment (hkf-tech) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Apprentice/ChangeOfEmployerViewModelMapper.cs]: changed GetLegalEntity to GetAccountLegalEntity :)

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Apprentice/ChangeOfEmployerViewModelMapper.cs [hunk 6]
```diff
@@ -5,38 +5,47 @@
...
-                    OldPrice = 0, // TODO add current apprenticeship cost and return in a field 'apprenticeship.Price or apprenticeship.Cost)
-                    NewEmployerName = "New Name", // TODO Determine where to Lookup the new account Name from? Do we call MA API? 
-                    NewStartDate = new MonthYearModel(source.StartDate),
-                    NewPrice = source.Price
+                    OldEmployerName = data.Apprenticeship.EmployerName,
+                    ApprenticeName = $"{data.Apprenticeship.FirstName} {data.Apprenticeship.LastName}",
+                    StopDate =   data.Apprenticeship.StopDate.Value, 
+                    OldStartDate = data.Apprenticeship.StartDate,
+                    OldPrice = decimal.ToInt32(data.PriceEpisodes.PriceEpisodes.GetPrice()), 
+                    NewEmployerName = data.AccountLegalEntity.LegalEntityName,
```
[the remark this answers is not in the document]
Comment (hkf-tech) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Mappers/Apprentice/ChangeOfEmployerViewModelMapper.cs]: Yes, you are correct. Fixed it.

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Views/Apprentice/ConfirmChangeOfEmployer.cshtml [hunk 7]
```diff
@@ -3,13 +3,25 @@
 @model SFA.DAS.ProviderCommitments.Web.Models.Apprentice.ChangeOfEmployerViewModel
 
 @{
-    ViewBag.Title = "Confirm the information before sending the your request";
+    ViewBag.Title = "Confirm the information before sending your request";
     ViewBag.GaData.Vpv = "/apprentices/change-employer/confirm";
```
[the remark this answers is not in the document]
Comment (hkf-tech) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Views/Apprentice/ConfirmChangeOfEmployer.cshtml]: updated

File: src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Views/Apprentice/ConfirmChangeOfEmployer.cshtml [hunk 8]
```diff
@@ -89,5 +101,5 @@
 
 @section breadcrumb
 {
-    <a id="back-link" class="govuk-back-link" href="@Url.RouteUrl(RouteNames.ApprenticePrice, new { Model.EmployerAccountLegalEntityPublicHashedId, Model.NewStartDate.MonthYear})">Back</a>
+    <a id="back-link" class="govuk-back-link" href="@Url.RouteUrl(RouteNames.ApprenticePrice, new { Model.EmployerAccountLegalEntityPublicHashedId, StartDate = Model.NewStartDate.MonthYear,Price = Model.NewPrice})">Back</a>
```
[the remark this answers is not in the document]
Comment (hkf-tech) [reply src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Views/Apprentice/ConfirmChangeOfEmployer.cshtml]: removed.

**Pull request merged**